### PR TITLE
Add Translate v3 Samples 

### PR DESCRIPTION
**Pull request by nnegrey**
Waiting for library release. 
(moved from tmp-generated-samples that Mike and I worked on)

### CODE REVIEW

**Comment by nnegrey** [comment]
Waiting on: https://github.com/googleapis/java-translate/pull/4

**Comment by nnegrey** [comment]
DO_NOT_DELETE BRANCH

**Comment by kurtisvg** [comment]
Can we try avoid opening PRs until they are ready to review? Github has a the ability to do [draft prs ](https://github.blog/2019-02-14-introducing-draft-pull-requests/) that won't trigger requests until they are ready. Alternatively, pushing to a personal fork works as well.

**Comment by nnegrey** [comment]
Yep, my bad.

**Comment by nnegrey** [comment]
@kurtisvg this is ready for review now. 
The test flakiness should be good now.

File: translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+              .addTargetLanguageCodes(targetLanguage)
+              .addInputConfigs(inputConfig)
+              .setOutputConfig(outputConfig)
+              .build();
+
+      OperationFuture<BatchTranslateResponse, BatchTranslateMetadata> future =
+          client.batchTranslateTextAsync(request);
+
+      System.out.println("Waiting for operation to complete...");
+      BatchTranslateResponse response = future.get();
```
Comment (kurtisvg) [inline translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java]: Any idea on how this fails? Is that when a `ExecutionException` is thrown?

File: translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      LocationName parent = LocationName.of(projectId, location);
+      GcsSource gcsSource = GcsSource.newBuilder().setInputUri(inputUri).build();
+      // Optional. Can be "text/plain" or "text/html".
+      InputConfig inputConfig =
+          InputConfig.newBuilder().setGcsSource(gcsSource).setMimeType("text/plain").build();
+      GcsDestination gcsDestination =
+          GcsDestination.newBuilder().setOutputUriPrefix(outputUri).build();
+      OutputConfig outputConfig =
+          OutputConfig.newBuilder().setGcsDestination(gcsDestination).build();
+      BatchTranslateTextRequest request =
```
Comment (kurtisvg) [inline translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java]: nit: These blocks feels a bit long to parse at once - might be better to add a newline to group some related declarations together.

File: translate/cloud-client/src/test/java/com/example/translate/QuickstartSampleIT.java [hunk 3]
```diff
@@ -35,6 +37,19 @@
   private ByteArrayOutputStream bout;
   private PrintStream out;
 
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+            System.getenv(varName),
```
Comment (kurtisvg) [inline translate/cloud-client/src/test/java/com/example/translate/QuickstartSampleIT.java]: We realized this in another PR, but we've been doing this backwards. It's `assertNotNull(Message, ObjectToAssert)`.

File: translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for Translate Text sample. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
```
Comment (kurtisvg) [inline translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java]: nit: I know we are doing this in other places, but I think going forward we should just us "Tests" over "IT".

File: translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  public void testTranslateText() throws IOException {
+    // Act
+    TranslateText.translateText(PROJECT_ID, "global", "sr-Latn", "Hello world");
+
+    // Assert
+    String got = bout.toString();
+
+    int count = 0;
+    if (got.contains("Zdravo svet")) {
+      count = 1;
```
Comment (kurtisvg) [inline translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java]: These are both assignments, not increments. 

If the goal is that the string contains at least one of these, just use something like:
```
assertThat(got, oneOf("string1", 'string2"));
```

If the goal is one, but not both then do something iterable (so future phrases can be easily added if it changes):

```java
List<String> acceptedResponses = List.of("Zdravo svet", "Pozdrav svijetu");
assertThat(accepted.stream().filter(r -> got.contains(r)).count()).isEqualto(1);
```

File: translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() throws InterruptedException, ExecutionException, IOException {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
```
Comment (kurtisvg) [inline translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java]: A comment on complicated setUp's like these detailing what it is trying to accomplish would be nice.

File: translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.concurrent.ExecutionException;
+
+public class BatchTranslateText {
+
+  public static void batchTranslateText()
+      throws InterruptedException, ExecutionException, IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "[Google Cloud Project ID]";
+    String location = "us-central1";
+    String sourceLanguage = "en";
```
[the remark this answers is not in the document]
Comment (nnegrey) [reply translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java]: I can't find documentation pages for location.
But I believe just `global` and `us-central1` are supported where if you use a custom model, you have to use `us-central1`. 

I'll inline `global` for samples that don't use a custom model and `us-central1` for those that do or I can add a comment:
```
# Supported Locations: `global` and `us-central`, `us-central1` is required if you are using a custom model.
String location = "global";
```

[the remark this answers is not in the document]
Comment (nnegrey) [reply translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java]: Well actually it is where ever AutoML supports as a location. But they don't have a page yet and so far it is only us-central1 but that could get out of date. 

Supported Locations: `global`, if using a custom model, you must use the same location that your model is hosted in. (`us-central1`)

File: translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java (hunk 4 above)
Comment (nnegrey) [reply translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java]: Change file names to `TranslateTextTests`?

Comment (kurtisvg) [reply translate/cloud-client/src/test/java/com/example/translate/TranslateTextIT.java]: SGTM

File: translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java (hunk 6 above)
Comment (nnegrey) [reply translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java]: How would you feel if I just called the samples for these `setup` / `teardown` functions?

```
CreateGlossary.createGlossary(
        PROJECT_ID, "us-central1", GLOSSARY_ID, languageCodes, GLOSSARY_INPUT_URI);
```

Most of these are just doing the same thing as the samples, just without calling them directly. It'd be easier to maintain, but then relies on the samples themselves and I know some languages prefer not to do this. 

Create a glossary and then delete it afterwards.

Comment (kurtisvg) [reply translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java]: I'm +1 to relying on the samples.

File: translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (kurtisvg) [reply translate/cloud-client/src/main/java/com/example/translate/BatchTranslateText.java]: Let's inline and use a comment on that then

**Pull request closed without merging**